beacon-chain/db/filesystem/blob.go

```diff
@@ -0,0 +1,170 @@
+package filesystem
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	ssz "github.com/prysmaticlabs/fastssz"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/verification"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/spf13/afero"
+)
+
+var (
+	errIndexOutOfBounds = errors.New("blob index in file name > MaxBlobsPerBlock")
+)
+
+const (
+	sszExt       = "ssz"
+	partExt      = "part"
+	blobLockPath = "blob.lock"
+)
+
+// NewBlobStorage creates a new instance of the BlobStorage object. Note that the implementation of BlobStorage may
+// attempt to hold a file lock to guarantee exclusive control of the blob storage directory, so this should only be
+// initialized once per beacon node.
+func NewBlobStorage(base string) (*BlobStorage, error) {
+	base = path.Clean(base)
+	fs := afero.NewBasePathFs(afero.NewOsFs(), base)
+	return &BlobStorage{fs: fs}, nil
+}
+
+// BlobStorage is the concrete implementation of the filesystem backend for saving and retrieving BlobSidecars.
+type BlobStorage struct {
+	fs afero.Fs
+}
+
+// Save saves blobs given a list of sidecars.
+func (bs *BlobStorage) Save(sidecar blocks.VerifiedROBlob) error {
+	fname := namerForSidecar(sidecar)
+	sszPath := fname.ssz()
```

Comment:
`.ssz()` returns the path, but from the first look it made it seems like it's returning the ssz object itself. Dont feel strongly about this, but it may be more straightforward just to name it `.path()`, and partial for `.patialPath()`

Reply:
https://github.com/prysmaticlabs/prysm/pull/13190/commits/6239576594a5bf3876cf00ac39da8a96a439ee8d